packages/runner/lib/sdk/locks.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable @typescript-eslint/require-await */
+import type { Result } from '@nangohq/utils';
+import { Err, Ok } from '@nangohq/utils';
+
+interface Lock {
+    key: string;
+    owner: string;
+    expiresAt: Date;
+}
+
+export class Locks {
+    private store = new Map<string, Lock>();
+    private mutex = new Map<string, boolean>();
```

Comment:
locks for the locks 😬

Reply:
the race condition is just below, between the `get` and `set` (if I was using pg, this read and write would be wrapped in a SELECT FOR UPDATE  transaction)
```
const existing = this.store.get(key);
if (...) {
...
}
this.store.set(key, { key, owner, expiresAt: new Date(now.getTime() + ttlMs) });
```

Reply:
I don't think it can happen right now, the code is synchronous, so the whole function is executed at once but maybe later with async indeed